zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/Commands.java

```diff
@@ -573,5 +574,27 @@ public CommandResponse run(ZooKeeperServer zkServer, Map<String, String> kwargs)
         }
     }
 
+    /**
+     * Returns the current ensemble configuration information.
+     * It provides list of current voting members in the ensemble.
+     */
+    public static class VotingViewCommand extends CommandBase {
+        public VotingViewCommand() {
+            super(Arrays.asList("voting_view"));
+        }
+
+        @Override
+        public CommandResponse run(ZooKeeperServer zkServer, Map<String, String> kwargs) {
+            CommandResponse response = initializeResponse();
+            if (zkServer instanceof QuorumZooKeeperServer) {
+                QuorumPeer peer = ((QuorumZooKeeperServer) zkServer).self;
+                response.put("current_config", peer.getVotingView());
```

Comment:
It would be better to create an explicit bean for the representation on this endpoint (that is really like a JSON based API) 
If we change the internal class we will change this API without notice.

Thoughts?

Reply:
That's a legitimate question and I don't know if I have the certain answer.

My thoughts are that jmx is clearly awesome at two things:
1. exposing internal feature flags and settings for review and alteration and
2. the execution of commands that require parameters.

The admin command is clearly awesome at one thing:
1. presenting complicated diagnostic information.

This feature easily falls into the second category (jmx is certainly not going to alter the voting view). I don't know enough about the history of the admin command to know if the division I outline above is the same as the one seen by the community. 

The other thing on my mind is that some users of ZooKeeper will opt out of jmx entirely because of the security concerns about the library. It would be generous to not shut them and their managing frameworks out of certain auditing information.